### PR TITLE
test: enable more esbuild dce tests

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**
more dce tests case

### CODE REVIEW

File: crates/rolldown/tests/esbuild/dce/.const_value_inlining_bundle/artifacts.snap.new
Comment (hyf0) [inline crates/rolldown/tests/esbuild/dce/.const_value_inlining_bundle/artifacts.snap.new:1]: I think it's ok to keep `.snap` in ignored fixtures, but `.new` files should be removed.

Comment (IWANABETHATGUY) [reply crates/rolldown/tests/esbuild/dce/.const_value_inlining_bundle/artifacts.snap.new:1]: still draft, not finished yet

Comment (IWANABETHATGUY) [reply crates/rolldown/tests/esbuild/dce/.const_value_inlining_bundle/artifacts.snap.new:1]: Done

**Comment by Boshen** [comment]
Is this using oxc?

**Review by hyf0 — APPROVED** [review]

**Comment by IWANABETHATGUY** [comment]
> Is this using oxc?

Some of the esbuild dec tests rely on `oxc` minifier, some are not.
These enabled test cases does not rely on `oxc` minifier

**Pull request merged**